lib/languages/js/source-code/source-code.js

```diff
@@ -868,6 +871,39 @@ class SourceCode extends TokenStore {
 		return ancestorsStartingAtParent.reverse();
 	}
 
+	/**
+	 * Determines whether the given identifier node is a reference to a global variable.
+	 * @param {ASTNode} node `Identifier` node to check.
+	 * @returns {boolean} True if the identifier is a reference to a global variable.
+	 */
+	isGlobalReference(node) {
+		if (!node) {
+			throw new TypeError("Missing required argument: node.");
+		}
+
+		if (node.type !== "Identifier") {
+			return false;
+		}
+
+		// If the identifier is a property in a MemberExpression, it's not a global reference
+		if (
+			node.parent.type === "MemberExpression" &&
+			node.parent.property === node &&
+			!node.parent.computed
+		) {
+			return false;
+		}
```

Comment:
Since we already add this shortcut, maybe worth to add more? 

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/3646efbd14979ca3dba438325aec2ffac33fa081/rules/ast/is-reference-identifier.js#L2

Reply:
I'm not even sure it's necessary to check for a MemberExpression. Any `Identifier` node that doesn't exist in a scope will return false. The only reason to check before calling `findVariable` is if `findVariable` is very slow. 

We already are using code without a lot of additional checks, so my feeling is that they're not needed. @snitin315 what do you think?

Reply:
To clarify, rules that currently use this logic are checking identifiers in specific positions, like e.g., new/call expression callees, where they can only be variable references. For a generic method that can take any node, we shouldn't assume that.

Reply:
Yes!